### PR TITLE
refactor/(iterable-batching): revamp batching iterable

### PR DESCRIPTION
**Pull request by sauravlal-rudderstack**
## Description of the change

> revamp batching of iterable

## Type of change
- [ ] Bug fix (non-breaking change that fixes an issue)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Related issues

> Fix [#1]() 

### Development

- [ ] Lint rules pass locally
- [ ] The code changed/added as part of this pull request has been covered with tests
- [ ] All tests related to the changed code pass in development

### Code review 

- [ ]  This pull request has a descriptive title and information useful to a reviewer. There may be a screenshot or screencast attached
- [ ] "Ready for review" label attached to the PR and reviewers mentioned in a comment
- [ ] Changes have been reviewed by at least one other engineer
- [ ] Issue from task tracker has a link to this pull request

### CODE REVIEW

File: v0/destinations/iterable/transform.js [hunk 1]
```diff
@@ -513,12 +477,21 @@ const processRouterDest = async inputs => {
...
   let identifyBatchedResponseList = [];
-  if (identifyArrayChunks.length) {
+  if (identifyEventChunks.length) {
+    // arrayChunks = [[e1,e2,e3,..batchSize],[e1,e2,e3,..batchSize]..]
+    // transformed payload of (n) batch size
+    const identifyArrayChunks = _.chunk(
+      identifyEventChunks,
+      IDENTIFY_MAX_BATCH_SIZE
+    );
     identifyBatchedResponseList = await batchEvents(identifyArrayChunks);
```
Comment (ItsSudip) [inline v0/destinations/iterable/transform.js:487]:
```suggestion
    identifyBatchedResponseList = batchEvents(identifyArrayChunks);
```
`batchEvents` is not an async function so await is not required.

File: v0/destinations/iterable/transform.js [hunk 2]
```diff
@@ -513,12 +477,21 @@ const processRouterDest = async inputs => {
...
   }
   // batching TrackArrayChunks
   let trackBatchedResponseList = [];
-  if (trackArrayChunks.length) {
+  if (trackEventChunks.length) {
+    // arrayChunks = [[e1,e2,e3,..batchSize],[e1,e2,e3,..batchSize]..]
+    // transformed payload of (n) batch size
+    const trackArrayChunks = _.chunk(trackEventChunks, TRACK_MAX_BATCH_SIZE);
     trackBatchedResponseList = await batchEvents(trackArrayChunks);
```
Comment (ItsSudip) [inline v0/destinations/iterable/transform.js:495]:
```suggestion
    trackBatchedResponseList = batchEvents(trackArrayChunks);
```

**Review by ItsSudip — CHANGES_REQUESTED** [review]

File: v0/destinations/iterable/transform.js (hunk 2 above)
Comment (sauravlal-rudderstack) [reply v0/destinations/iterable/transform.js]: Updated

File: v0/destinations/iterable/transform.js (hunk 1 above)
Comment (sauravlal-rudderstack) [reply v0/destinations/iterable/transform.js]: Yeah, updated

**Review by ItsSudip — APPROVED** [review]

**Review by akashrpo — APPROVED** [review]

**Pull request merged**